ext/standard/file.c

```diff
@@ -181,14 +181,20 @@ PHP_MSHUTDOWN_FUNCTION(file) /* {{{ */
 }
 /* }}} */
 
+static inline bool php_is_valid_flock_flag(zend_long s) {
+	const zend_long sb = s & ~PHP_LOCK_NB;
+	return (sb == PHP_LOCK_UN || sb == PHP_LOCK_SH ||
+		sb == PHP_LOCK_EX || s == -1);
```

Comment:
what is `-1`?

Reply:
It seems to be an accepted value (a test uses it) even tough it does nothing. I may discard it.

Reply:
In my understanding `-1` flag implies all flags set, thus more than 1 exclusive flag set, thus it should not be allowed and hence this PR.